sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -194,6 +194,17 @@ case class UnknownPartitioning(numPartitions: Int) extends Partitioning {
   override def guarantees(other: Partitioning): Boolean = false
 }
 
+case class RandomPartitioning(numPartitions: Int) extends Partitioning {
```

Comment:
Actually, I change my mind: this is a fine name. The fact that we do the partitioning in a round-robin fashion is an implementation detail, not an inherent feature of the partitioning itself.